Feature/key anchoring

Carlos-Augusto opened the pull request.

lalottacontinua commented:
- class PubKeyService, method publish()  -> there is a .get called on an option; shouldn't it better be getOrElse?
- same in class KeyAnchroing, method anchorKey()
or is the earlyResponseIf() somehow interrupting the for-comprehension?
Loooks good otherwise, though I don't understand the naming of Wolf and Tiger.. ;)

Carlos-Augusto commented:
***
I have implemented an operator that allows to control the flow of the execution and of conditions throughout a for-comprehension. This operator is _earlyResponseIf_ (I have added others); by using this operator, it is possible to evaluate conditions and break out elegantly from within a for-comprehension. It also allows to fine-control the kinds of errors that can be presented during the execution.  
Having said this, the existence of the maybe-variables is being safely checked with these operators, therefore, it is safe to unbox the value at that point.
***
Software should have a meaning beyond a simple name. Code names if you will. I have named the producers like that to show this naming pattern. My goal is that it is clear, nice, and that the code still convey its meaning.
